src/tests/test_transform.py

```diff
@@ -96,23 +95,17 @@ def test_isAddonNewer(self):
 		newAddon.lastTestedVersion = V_2020_3
 		newAddon.pathToData = "new-path"
 
-		self.assertDictEqual(getLatestAddons([oldAddon, newAddon], nvdaAPIVersions), {
-			# 1st addon is added because: oldAddon.addonId not in addons
-			# 2nd addon it not added: pass as incompatible
-			V_2020_2: {"stable": {"foo": oldAddon}, "beta": {}},
-			# 1st addon is added because: oldAddon.addonId not in addons
-			# 2nd addon updates it because: newAddon.addonVersion > oldAddon.addonVersion
-			V_2020_3: {"stable": {"foo": newAddon}, "beta": {}},
+		addonGenerator = (addon for addon in (oldAddon, newAddon))
+		self.assertDictEqual(getLatestAddons(addonGenerator, nvdaAPIVersions), {
+			# For V_2020_3 newAddon overrides oldAddon because newAddon.addonVersion > oldAddon.addonVersion
+			V_2020_3: {"stable": {"foo": newAddon}, "beta": {}}
 		})
 
 		# Reverse order of addons
-		self.assertDictEqual(getLatestAddons([newAddon, oldAddon], nvdaAPIVersions), {
-			# 1st addon is not added because: pass as incompatible
-			# 2nd addon is added because: oldAddon.addonId not in addons
-			V_2020_2: {"stable": {"foo": oldAddon}, "beta": {}},
-			# 1st addon is added because: newAddon.addonId not in addons
-			# 2nd addon does not update it because: oldAddon.addonVersion < newAddon.addonVersion
-			V_2020_3: {"stable": {"foo": newAddon}, "beta": {}},
+		addonGenerator = (addon for addon in (newAddon, oldAddon))
+		self.assertDictEqual(getLatestAddons(addonGenerator, nvdaAPIVersions), {
+			# For V_2020_3 oldAddon does not override newAddon because newAddon.addonVersion > oldAddon.addonVersion
+			V_2020_3: {"stable": {"foo": newAddon}, "beta": {}}
```

Comment:
Since the goal of this test is to show that the output does not depend on the input, why not just compare them directly:
```python
# Ensure order of the addon list does not matter.
self.assertDictEqual(
	getLatestAddons([oldAddon, newAddon], nvdaAPIVersions),
	getLatestAddons([newAddon, oldAddon], nvdaAPIVersions)
)
# For ease of development, verify output
self.assertDictEqual(
	getLatestAddons([newAddon, oldAddon], nvdaAPIVersions),
	V_2020_3: {"stable": {"foo": newAddon}, "beta": {}}
)

Reply:
I don't think this says "why" the test is occurring, what the test aims to achieve. Why should the order of the addon list not matter? Because we want to assert that the newest addon is always used, and this exercises the two main pathways in `_isAddonNewer`.

Reply:
Yes, you could add a comment to the example say, "output relies entirely on the relative properties of the addons, not their order". But I'd say that aspect is more obvious in one of the other tests.